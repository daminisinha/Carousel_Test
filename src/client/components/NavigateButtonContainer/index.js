import React, { Component } from 'react';
import { Visible } from 'react-grid-system';
import NavigateButton from '../NavigateButton';
import './index.scss'
class NavigateButtonContainer extends Component {

    render() {
        return (
            <ul className="desktop-navigate-buttons">
            <li className="button-navigate border-prev-button"><Visible lg xl>
                     <NavigateButton onClickHandler={this.props.prevImage} buttonText="Prev"/>
                </Visible></li>
            <li className="button-navigate border-next-button"><Visible lg xl>
                     <NavigateButton   onClickHandler={this.props.nextImage} buttonText="Next"/>
                </Visible></li>
        </ul>
        );
    }
}

export default NavigateButtonContainer;