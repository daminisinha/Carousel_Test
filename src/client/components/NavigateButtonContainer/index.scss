@import '../../styles/variables.scss';

.desktop-navigate-buttons{
    list-style-type: none;
    display:flex;
    justify-content: center;
    padding-left:0px;
    margin-top:0px;
}
.button-navigate>button{
    background-color:$app-color;
    color:white;
    width:100px;
    height:35px;
    font-size:14px;
}

.border-prev-button>button{
    border-top-left-radius: 25px;
    border-bottom-left-radius:25px;
}

.border-next-button>button{
    border-top-right-radius: 25px;
    border-bottom-right-radius:25px;
}