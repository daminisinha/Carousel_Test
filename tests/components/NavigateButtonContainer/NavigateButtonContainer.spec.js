import React from 'react';
import NavigateButtonContainer from '../../../src/client/components/NavigateButtonContainer'
import {shallow} from 'enzyme';

test('Navigate Button container renders',()=>{
    const component = shallow(<NavigateButtonContainer/>); 
    expect(component).toMatchSnapshot();
})
