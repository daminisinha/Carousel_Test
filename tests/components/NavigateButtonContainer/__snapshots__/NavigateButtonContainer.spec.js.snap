// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigate Button container renders 1`] = `
<ul
  className="desktop-navigate-buttons"
>
  <li
    className="button-navigate border-prev-button"
  >
    <Visible
      lg={true}
      md={false}
      sm={false}
      xl={true}
      xs={false}
    >
      <NavigateButton
        buttonText="Prev"
      />
    </Visible>
  </li>
  <li
    className="button-navigate border-next-button"
  >
    <Visible
      lg={true}
      md={false}
      sm={false}
      xl={true}
      xs={false}
    >
      <NavigateButton
        buttonText="Next"
      />
    </Visible>
  </li>
</ul>
`;
